app/controllers/admin/base_controller.rb

```diff
@@ -0,0 +1,17 @@
+class Admin::BaseController < ApplicationController
+
+  before_action :require_admin
+
+  def require_admin
+    render file: "/public/404" unless current_admin?
+  end
+
+  def dashboard
+    render 'admin/dashboard'
+  end
+
+  def current_admin?
+    current_user && current_user.admin?
```

Comment:
Should current_admin live in the application controller? Something to keep in mind as we build functionality.